xgcm/grid.py

```diff
@@ -1255,34 +1258,62 @@ def check_neighbor(link, position):
+            )
+
+        # initialize empty list
+        self._metrics[metric_axes] = []
+        for metric_varname in _maybe_promote_str_to_list(value):
+            if metric_varname not in self._ds:
                 raise KeyError(
-                    "Metric axes %r not compatible with grid axes %r"
-                    % (metric_axes, tuple(self.axes))
+                    f"Metric variable {metric_varname} not found in dataset."
                 )
-            # initialize empty list
-            self._metrics[metric_axes] = []
-            for metric_varname in _maybe_promote_str_to_list(metric_vars):
-                if metric_varname not in self._ds:
-                    raise KeyError(
-                        "Metric variable %s not found in dataset." % metric_varname
-                    )
-                # resetting coords avoids potential broadcasting / alignment issues
-                metric_var = self._ds[metric_varname].reset_coords(drop=True)
-                # TODO: check for consistency of metric_var dims with axis dims
-                # check for duplicate dimensions among each axis metric
-                self._metrics[metric_axes].append(metric_var)
+            # resetting coords avoids potential broadcasting / alignment issues
+            metric_var = self._ds[metric_varname].reset_coords(drop=True)
+
+            # TODO: check for consistency of metric_var dims with axis dims
+            # check for duplicate dimensions among each axis metric
+            self._metrics[metric_axes].append(metric_var)
+
+    def _assign_metrics(self, metrics):
+        pass
+        # """
+        # metrics should look like
+        #    {('X', 'Y'): ['rAC']}
+        # check to make sure everything is a valid dimension
+        # """
+
+        # self._metrics = {}
+
+        # for key, metric_vars in metrics.items():
+        #     metric_axes = frozenset(_maybe_promote_str_to_list(key))
+        #     if not all([ma in self.axes for ma in metric_axes]):
+        #         raise KeyError(
+        #             "Metric axes %r not compatible with grid axes %r"
+        #             % (metric_axes, tuple(self.axes))
+        #         )
+        #     # initialize empty list
+        #     self._metrics[metric_axes] = []
+        #     for metric_varname in _maybe_promote_str_to_list(metric_vars):
+        #         if metric_varname not in self._ds:
+        #             raise KeyError(
+        #                 "Metric variable %s not found in dataset." % metric_varname
+        #             )
+        #         # resetting coords avoids potential broadcasting / alignment issues
+        #         metric_var = self._ds[metric_varname].reset_coords(drop=True)
+        #         # TODO: check for consistency of metric_var dims with axis dims
+        #         # check for duplicate dimensions among each axis metric
+        #         self._metrics[metric_axes].append(metric_var)
```

Comment:
Is this still needed?

Reply:
You're right, I think set_metrics already works, so I have removed this section.